genie-core/src/main/java/com/netflix/genie/core/services/impl/S3FileTransferImpl.java

```diff
@@ -71,8 +73,13 @@ public S3FileTransferImpl(@NotNull final AmazonS3 amazonS3Client, @NotNull final
     @Override
     public boolean isValid(final String fileName) throws GenieException {
         log.debug("Called with file name {}", fileName);
-        final Matcher matcher = this.s3PrefixPattern.matcher(fileName);
-        return matcher.matches();
+        try {
+            getS3Uri(fileName);
+            return true;
+        } catch (GenieServerException e) {
+            log.error("Invalid S3 path {} ({})", fileName, e.getMessage());
+        }
+        return false;
```

Comment:
Seems like you should just put the false inside the catch as there is no other way to out of it